trin-core/src/portalnet/find/query_trace.rs

```diff
@@ -0,0 +1,219 @@
+use crate::portalnet::Enr;
+use crate::utils::bytes::hex_encode;
+
+use discv5::enr::NodeId;
+use serde::Serialize;
+// use ssz_types::typenum::Xor;
+use std::collections::HashMap;
+use std::time::{Duration, SystemTime};
+
+use crate::portalnet::types::distance::{Metric, XorMetric};
+
+///Store ENRs as strings for hashability
+type NodeIdString = String;
+
+type ContentId = NodeId;
+
+/// Keeps track of query details.
+#[derive(Debug, Serialize)]
+pub struct QueryTrace {
+    /// None if the content was not found.
+    pub received_content_from_node: Option<NodeIdString>,
+    /// The local node.
+    pub origin: NodeIdString,
+    /// Map of a node's ID to its corresponding `QueryResponse`
+    pub responses: HashMap<NodeIdString, QueryResponse>,
+    pub node_metadata: HashMap<NodeIdString, NodeInfo>,
```

Comment:
From above: 

> Also note that Node IDs are now being used as the primary ID for each node rather than ENRs. This was due to a bug caused by multiple ENRs for the same node ID making it into the QueryTrace.

Reply:
I assume that we don't use `NodeId` because it doesn't implement the `serde` traits then? I would prefer we convert at the JSON-RPC layer as opposed to making the internal logic less straightforward.